### PR TITLE
Implementation of `one-or-more` rule

### PR DESCRIPTION
**Pull request by ejoebstl**
Properly implements a rule for RepOnce, instead of translating it to `x ~ x*`. This solves the whitespace issue described in #396.

### CODE REVIEW

**Comment by dragostis** [comment]
Thanks for putting this together! As I said, I'm afraid that this might break people, but I'm willing to try to see if dependent crates pass their tests or not with this fix.

**Comment by NoahTheDuke** [comment]
Any chance this could get updated?

File: meta/src/optimizer/mod.rs [hunk 1]
```diff
@@ -105,6 +104,7 @@ pub enum OptimizedExpr {
     Choice(Box<OptimizedExpr>, Box<OptimizedExpr>),
     Opt(Box<OptimizedExpr>),
     Rep(Box<OptimizedExpr>),
+    RepOnce(Box<OptimizedExpr>),
```
Comment (tomtau) [inline meta/src/optimizer/mod.rs:107]: `enum OptimizedExpr` wasn't annotated with `#[non_exhaustive]`, so adding a new variant to it is unfortunately a semver breaking change

**Review by tomtau — COMMENTED** [review]
the fix seems all right, but it's a breaking change: both logic-wise (if any external crates depended on the old behaviour) and API-wise due to a new variant on a public enum, so potentially something for 3.x

**Comment by tomtau** [comment]
revived this PR in https://github.com/pest-parser/pest/pull/878

**Pull request closed without merging**